packages/wallet/frontend/src/pages/kyc.tsx

```diff
@@ -0,0 +1,120 @@
+import { HeaderLogo } from '@/components/HeaderLogo'
+import AuthLayout from '@/components/layouts/AuthLayout'
+import { userService } from '@/lib/api/user'
+import { NextPageWithLayout } from '@/lib/types/app'
+import {
+  GateHubMessageType,
+  type GateHubMessageError
+} from '@/lib/types/windowMessages'
+import { GetServerSideProps, InferGetServerSidePropsType } from 'next/types'
+import { useEffect } from 'react'
+
+type KYCPageProps = InferGetServerSidePropsType<typeof getServerSideProps>
+
+type MessageData =
+  | {
+      type: GateHubMessageType.OnboardingCompleted
+      value: 'submitted' | 'resubmitted'
+    }
+  | { type: GateHubMessageType.OnboardingError; value: GateHubMessageError }
+  | { type: GateHubMessageType.OnboardingInitialized }
+
+const KYCPage: NextPageWithLayout<KYCPageProps> = ({
+  url,
+  addUserToGatewayUrl
+}) => {
+  // const [openDialog, closeDialog] = useDialog()
+  // const router = useRouter()
+
+  useEffect(() => {
+    // TODO: Handle the received message from iframe
+    // https://docs.gatehub.net/api-documentation/c3OPAp5dM191CDAdwyYS/gatehub-products/gatehub-onboarding#message-events
+    const onMessage = async (e: MessageEvent<MessageData>) => {
+      console.log('received message from iframe', { e })
```

Comment:
having some logs will help debugging in case there are some issues with the messages from iframe
I will let @Tymmmy and @raducristianpopa to decide what is best to do here

Reply:
We can leave it like this for now, and go ahead with the PR approval and merging.